static/js/containers/SignupDialog_test.js

```diff
@@ -37,16 +37,38 @@ describe("SignupDialog", () => {
     helper.store.dispatch(setDialogVisibility(true));
     renderDialog();
 
-    let link = document.body.querySelector(".signup-dialog a");
+    const link = document.body.querySelector(".signup-dialog a");
     assert.equal(link.getAttribute('href'), `/login/edxorg${queryParams}`);
   });
 
+  it('also checks the coupon query param', () => {
+    window.location = "http://fake/?coupon=whee";
+    helper.store.dispatch(setDialogVisibility(true));
+    renderDialog();
+
+    const link = document.body.querySelector(".signup-dialog a");
+    const expectedNext = "/dashboard/?coupon=whee";
```

Comment:
According to [the URLconf for the UserCouponsView](https://github.com/mitodl/micromasters/blob/master/ecommerce/urls.py#L14), the only allowable characters for a coupon are letters, numbers, dashes, and periods. I've changed the code and the test to do all URL encoding using `urijs`, to make sure that any edge cases are handled by that library.